README.md

```diff
@@ -55,39 +55,80 @@ page.to_h
+class FileOnDiskRequest < Scraped::Request::Strategy
+  def response
+    Response.new(url: url, body: open(filename).read)
+  end
+
+  private
+
+  def filename
+    @filename ||= File.join(URI.parse(url).host, Digest::SHA1.hexdigest(url))
+  end
+end
 ```
 
-This will use the [`scraped_page_archive`](https://github.com/everypolitician/scraped_page_archive)
-gem to store a copy of the pages you scrape in a git branch in your scraper's repo.
+The `response` method should return a `Response` instance. You need to pass at least `url` and `body` parameters to the `Response` constructor, you can also pass `status` and `headers` parameters.
+
+To use a custom request strategy pass it to `Scraped::Request`:
 
-## Custom strategies
+```ruby
+request = Scraped::Request.new(url: 'http://example.com', strategies: [FileOnDiskRequest, Scraped::Request::Strategy::LiveRequest])
+page = MyPersonPage.new(response: request.response)
+```
 
-If for some reason you can't scrape a site using one of the built in strategies
-then you can provide your own strategy.
+### Decorated responses
 
-A strategy is an object that implements a `response` method which returns a
-`Hash` with at least a `:body` key and optionally `:status` and `:headers` keys.
+To manipulate the response before it is passed to the scraper create a class that subclasses `Scraped::Response::Decorator`:
 
 ```ruby
-class FilesystemStrategy
-  def response(url)
-    { body: File.read(Digest::SHA1.hexdigest(url) + '.html') }
+class AbsoluteLinks < Scraped::Response::Decorator
+  def body
+    doc = Nokogiri::HTML(response.body)
+    doc.css('a').each do |link|
+      link[:href] = URI.join(response.url, link[:href]).to_s
+    end
+    doc.to_s
   end
 end
 ```
 
-Then you can pass that strategy when creating an instance of your `Scraped` subclass:
+As well as the `body` method you can also supply your own `url`, `status` and `headers` methods. You can access the original response instance via the `response` method, as with `response.body` and `response.url` above.
+
+To use a response decorator you need to supply it to the `Request#response` method:
+
+```ruby
+request = Scraped::Request.new(url: 'http://example.com')
+response = request.response([AbsoluteLinks])
+```
+
+### Configurating requests and responses
+
+When passing an array of request strategies or response decorators you should always pass the class, rather than the instance. If you want to configure an instance you can pass in a two element array where the first element is the class and the second element is the config:
 
 ```ruby
-ExamplePage.new(url: 'http://example.com', strategy: FilesystemStrategy.new)
+class CustomHeader < Scraped::Response::Decorator
+  def headers
+    response.headers.merge('X-Greeting' => config[:greeting])
+  end
+end
+
+response = Scraped::Request.new(url: url).response([
+  [CustomHeader, greeting: 'Hello, world']
+])
```

Comment:
I'm open to persuasion on this, but this feels quite ugly and kludgy to me. Perhaps better to make this a hash?

```ruby
response = Scraped::Request.new(url: url).response([
   { decorator: CustomHeader, greeting: 'Hello, world' }
 ])
```